src/Event.h

```diff
@@ -121,6 +154,22 @@ class EventMgr final : public Obj, public iosource::IOSource {
      */
     const Event* CurrentEvent() { return current; }
 
+    /**
+     * Register a script-layer metadata identifier *id* with type *type*.
+     *
+     * @param id The script-level ``EventMetadata::ID`` enum value.
+     * @param type The type to expect for the given metadata identifier.
+     */
+    bool RegisterMetadata(EnumValPtr id, TypePtr type);
```

Comment:
Argh, I think the better place for the metadata type tracking is in `EventRegistry`. Will move there after the first round of review.